infrastructure/src/test/java/org/corfudb/infrastructure/log/statetransfer/DelayedExecution.java

```diff
@@ -0,0 +1,28 @@
+package org.corfudb.infrastructure.log.statetransfer;
+
+import org.corfudb.infrastructure.log.statetransfer.batch.BatchResult;
+import org.corfudb.util.CFUtils;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.function.Supplier;
+
+public interface DelayedExecution {
+
+    default CompletableFuture<BatchResult> withDelayOf(Supplier<CompletableFuture<BatchResult>> result,
+                                                       long delayInMillis,
+                                                       ScheduledExecutorService executor) {
+        try {
+            return CFUtils.runFutureAfter(result, executor, delayInMillis);
```

Comment:
please try to avoid using CFUtils

Reply:
I wrote this function in CFUtils to aid with a delayed execution.

Reply:
sorry, didn't notice. I thought it's `waitUninteruptibly` method